### PR TITLE
[Token] update token indexer

### PR DESCRIPTION
**Pull request by areshand**
### Description

1. add mutate_token_property entry function in SDK and add smoke-test for mutating token properties
2. update token indexer: (1) update the event ingestion to use new events, (2 ) add option for turning on/off for URI data fetcher, (3) add token txn in indexer e2e test
### Test Plan
1 add new e2e test for token indexer
2 manaully test locally and see data populated as expected

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2467)

### CODE REVIEW

File: testsuite/smoke-test/src/nft_transaction.rs [hunk 1]
```diff
@@ -50,7 +50,7 @@ impl AptosTest for NFTTransaction {
                     creator.address(),
                     0,
                     0,
```
Comment (davidiw) [inline testsuite/smoke-test/src/nft_transaction.rs:52]: should move this to a move e2e test

**Review by davidiw — COMMENTED** [review]
just to be clear you actually started up your own local indexer?

wow

gonna leave open just for someone else to see it -- but if no other feedback, I will accept

File: testsuite/smoke-test/src/nft_transaction.rs (hunk 1 above)
Comment (areshand) [reply testsuite/smoke-test/src/nft_transaction.rs:52]: let's keep the scope of this PR as it is and tackle the test migration in a separate PR.
1. we need to move the entire NFT smoke-test to move-e2e test, 2 we need to use sdk builder there as well.

File: ecosystem/indexer/migrations/2022-08-02-204644_token_propertys/up.sql [hunk 2]
```diff
@@ -0,0 +1,14 @@
+-- Your SQL goes here
+CREATE TABLE token_propertys
```
Comment (bowenyang007) [inline ecosystem/indexer/migrations/2022-08-02-204644_token_propertys/up.sql:2]: Ideally rename to token_properties.

**Comment by bowenyang007** [comment]
@davidiw what do you mean by starting your own local indexer? As far as I see this looks good (until we complete migration to firehose of course).

**Review by bowenyang007 — APPROVED** [review]
token_properties instead of token_propertys. Otherwise looks good to me.

File: ecosystem/indexer/migrations/2022-08-02-204644_token_propertys/up.sql (hunk 2 above)
Comment (areshand) [reply ecosystem/indexer/migrations/2022-08-02-204644_token_propertys/up.sql:2]: I tried to name it token_properties and the diesel fail to recognize the name.  Diesel sucks and unfortunately I cannot help.

File: testsuite/smoke-test/src/nft_transaction.rs (hunk 1 above)
Comment (areshand) [reply testsuite/smoke-test/src/nft_transaction.rs:52]: Yes, started a local node and send some dummy TXNs. I also started a local indexer tailing from my local node to test the NFT events parsing and see the data populated in my local postgres DB

**Pull request merged**